docs/user/sampling.md

```diff
@@ -40,34 +40,76 @@ on a single CPU. `VecWorker` can compute a batch of actions from a policy
 regarding multiple environments to reduce of overhead of sampling (e.g. feeding
 forward a neural network).
 
-## Setup Sampler and Worker for a Trainer
+## Construct a Sampler
 
-Setup the sampler and worker for a `Trainer` is easy. Just passing
-`sampler_cls` and `worker_class` to `trainer.setup()`. The number of workers in
-the sampler can be set by the parameter `n_workers`.
+A sampler can be constructed either from a worker factory (the class that can
+construct workers), or from parameters directly.
 
-For `VecWorker`, you can set the level of vectorization (i.e. the number of
-environments simulated in one step) by setting `n_envs` in `worker_args`.
+### From a Worker Factory
+
+```py
+from garage.sampler import LocalSampler, WorkerFactory
+
+env = ...
+policy = ...
+worker_factory = WorkerFactory(max_episode_length=100,
+                               is_tf_worker=True,
+                               n_workers=4)
+sampler = LocalSampler.from_worker_factory(worker_factory=worker_factory,
+                                           agents=policy,
+                                           envs=env)
+...
+```
+
+In the above example, we firstly construct a worker factory, which will
+construct 4 workers for the sampler. And the max length of episodes collected
+by these workers will be 100. Noted that for policies with TensorFLow
+framework, we need to set `is_tf_worker` to be `True`. Here we don't choose
+a type of worker explicitly, so it will construct `DefaultWorker` by default.
+
+With the worker factory, then we construct a `LocalSampler` with policies and
+environments that will be used in sampling.
+
+### From parameters
```

Comment:
Can you actually swap the order of these two sections? It's now much more common to construct a sampler directly.